3D-Engine/Shader.cpp

```diff
@@ -0,0 +1,120 @@
+#include <iostream>
+#include <fstream>
+#include "Shader.h"
+
+
+Shader::Shader(const char* vertexPath, const char* fragmentPath)
+{
+    std::string VertexSource = Load(vertexPath);
+    std::string FragmentSource = Load(fragmentPath);
+
+    GLuint VertexID = Compile(GL_VERTEX_SHADER, &VertexSource);
+    GLuint FragmentID = Compile(GL_FRAGMENT_SHADER, &FragmentSource);
+
+    m_ProgramID = glCreateProgram();
+    glAttachShader(m_ProgramID, VertexID);
+    glAttachShader(m_ProgramID, FragmentID);
+    glLinkProgram(m_ProgramID);
+
+    LinkError();
+}
+
+std::string Shader::Load(const char* path)
+{
+    std::ifstream file;
+    std::string source, line;
+
+    file.open(path, std::ios::in);
+
+    if (!file.is_open())
+    {
+        std::cerr << std::string(path) + std::string(" failed to open") << std::endl;
+        exit(-1);
+    }
+
+    while (std::getline(file, line))
+    {
+        source += line;
+        source += '\n';
+    }
+
+    file.close();
+
+    return source;
+}
+
+void Shader::CompileError(GLuint ShaderID)
+{
+    int return_code;
+    glGetShaderiv(ShaderID, GL_COMPILE_STATUS, &return_code);
+
+    if (return_code == GL_FALSE)
+    {
+        int length;
+
+        glGetShaderiv(ShaderID, GL_INFO_LOG_LENGTH, &length);
+
+        char* errlog = new char[length];
```

Comment:
new is considered bad practice, prefer to use smart pointers like `std::unique_ptr<char[]>`